src/werkzeug/datastructures.py

```diff
@@ -1755,6 +1768,46 @@ def to_header(self):
             result.append(value)
         return ",".join(result)
 
+    @t.overload
+    @classmethod
+    def parse_header(cls, value: t.Optional[str]) -> "Accept":
+        ...
+
+    @t.overload
+    @classmethod
+    def parse_header(cls, value: t.Optional[str]) -> _TAnyAccept:
+        ...
+
+    @classmethod
+    def parse_header(cls, value: t.Optional[str]) -> _TAnyAccept:
```

Comment:
I don't think this typing makes sense (the overloads and the `_TAnyAccept`) this classmethod will only return an Accept.